composer/algorithms/seq_length_warmup/seq_length_warmup.py

```diff
@@ -27,9 +27,33 @@ def set_batch_sequence_length(batch: Dict[str, Tensor], curr_seq_len: int, trunc
     .. note::
 
         The schedule for ``curr_seq_len`` over training time should be managed
-        out of this function.
+        outside of this function.
 
-    Example: Awaiting language model test fixtures.
+    .. note::
+
+        Variable input lengths can create CUDA OOM errors. To avoid this,
+        we follow `PyTorch notes <https://pytorch.org/tutorials/recipes/recipes/tuning_guide.html#pre-allocate-memory-in-case-of-variable-input-length>`_
+        and pre-allocate the memory with a blank forward and backward pass.
+
+    Example:
+    
+    .. code-block::
+
+        import composer.functional as cf
+
+        def training_loop(model, train_loader, sequence_length):
+            opt = torch.optim.Adam(model.parameters())
+            loss_fn = F.cross_entropy
+            model.train()
+
```

Comment:
could we rip out the outer training loop / shorten this a bit? I could be persuaded otherwise, but I'm used to seeing docstring examples that are the minimal runnable code to show correct calls and associated return values.

Reply:
right now the actual function call is buried in 90% boilerplate; could probably just stick in an otherwise-empty `for X, y in train_loader` loop and still get the same clarity.

Reply:
I was just using the method card template, but I agree and will do this!